packages/core/src/fhir-deduplication/resources/observation-social.ts

```diff
@@ -64,17 +75,27 @@ export function groupSameObservationsSocial(observations: Observation[]): {
     const keyCode = retrieveCode(keyCodes);
     const value = extractValueFromObservation(observation);
 
-    if (value && keyCode) {
-      const key = keyCode ? JSON.stringify({ value, keyCode }) : undefined;
-      if (key) {
-        fillMaps(observationsMap, key, observation, refReplacementMap, undefined, postProcess);
+    if (!value) {
+      danglingReferencesSet.add(createRef(observation));
+      continue;
+    }
+
+    let key;
+    if (keyCode) {
+      key = JSON.stringify({ value, keyCode });
+    } else {
+      const display = extractDisplayFromConcept(observation.code);
+      if (display) {
+        key = JSON.stringify({ value, display });
```

Comment:
we need an else here to put those with display undefined into dangling refs

Reply:
yup yup yup